From 272215415ba85e9e60324aae0d7b0b67bdecb72b Mon Sep 17 00:00:00 2001
From: tonishi92 <tatsuo.onishi@gmail.com>
Date: Tue, 6 Feb 2024 11:45:23 +0100
Subject: [PATCH 22/52] Add MOZART_MOSAIC in a condition on emiss_opt for
 Nightingale DMS emissions in chem/chemics_init.F

emiss_opt =  ecbmz_mosaic (4)
          or mozcem (8)
          or mozmem (10)
          or esaprcnov (13)
chem_opt  =  SAPRC99_MOSAIC_8BIN_VBS2_AQ_KPP (203)
          or MOZART_MOSAIC_4BIN_AQ_KPP (202)
          or CBMZ_MOSAIC_DMS_8BIN_AQ (34)
---
 chem/chemics_init.F | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/chem/chemics_init.F b/chem/chemics_init.F
index 333fcad5..acfeb9bd 100755
--- a/chem/chemics_init.F
+++ b/chem/chemics_init.F
@@ -487,9 +487,10 @@ call wrf_message("**************************************************************
    endif
 
 !lm Add this condition on emiss_opt for Nightingale DMS emissions
-   if ( config_flags%dmsemis_opt == 2 .and. (config_flags%emiss_opt /= 4 .and. config_flags%emiss_opt /= 13) ) then
+   if ( config_flags%dmsemis_opt == 2 .and. (config_flags%emiss_opt /= 4 .and. config_flags%emiss_opt /= 8 &
+                                          .and. config_flags%emiss_opt /= 10 .and. config_flags%emiss_opt /= 13) ) then
       call wrf_error_fatal (15,'chemics_init: error, nightingale_dmsemis (dmsemis_opt = 2) &
-                              requires e_dms_oc to be defined in emis_ant (emiss_opt = 4 or 13)')
+                              requires e_dms_oc to be defined in emis_ant (emiss_opt = 4, 8, 10 or 13)')
    endif
 !lm
 
@@ -559,11 +560,13 @@ call wrf_message("**************************************************************
 !lm For now, do not allow dmsemis_opt=2 for untested mechanisms.
     IF(config_flags%dmsemis_opt == 2) THEN
         IF(config_flags%chem_opt == SAPRC99_MOSAIC_8BIN_VBS2_AQ_KPP .or. &
+           config_flags%chem_opt == MOZART_MOSAIC_4BIN_AQ_KPP .or. &
            config_flags%chem_opt == CBMZ_MOSAIC_DMS_8BIN_AQ) THEN
            call wrf_debug( 15, 'Using the DMS emission scheme for MOSAIC')
         ELSE
            message_txt = "chemics_init.F: ERROR, DMSEMIS_OPT == 2 is currently only allowed with &
-                          tested mechanisms SAPRC99_MOSAIC_8BIN_VBS2_AQ_KPP and CBMZ_MOSAIC_DMS_8BIN_AQ.&
+                          tested mechanisms SAPRC99_MOSAIC_8BIN_VBS2_AQ_KPP, &
+                          MOZART_MOSAIC_4BIN_AQ_KPP and CBMZ_MOSAIC_DMS_8BIN_AQ.&
                           In order to use this option with another mechanism, please modify chemics_init.F"
            call wrf_error_fatal( trim(message_txt) )
         ENDIF
-- 
2.31.1

